### PR TITLE
PHOENIX-5597 No read repair happens when scans filter rows based on a…

### PR DESCRIPTION
**Pull request by kadirozde**
… covered column

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesOptions.java [hunk 1]
```diff
@@ -355,7 +355,7 @@
     public static final long DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS = 30*60*1000; // 30 min
     public static final long DEFAULT_TASK_HANDLING_INITIAL_DELAY_MS = 10*1000; // 10 sec
 
-    public static final long DEFAULT_GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS = 10*60*1000; /* 10 minutes */
+    public static final long DEFAULT_GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS = 7*24*60*60*1000; /* 7 days */
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesOptions.java]: How did we arrive on 7 days as the stopgap before deleting unverified rows? Is it related to the frequency of major compaction or just a general "enough time" to ensure replication lag is covered?

If the former, I would suggest to set this to the value set for the config HConstants.MAJOR_COMPACTION_PERIOD` or default to `HConstants.DEFAULT_MAJOR_COMPACTION_PERIOD`.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java [hunk 2]
```diff
@@ -290,27 +290,6 @@ public static void checkUnverifiedCellCount(Connection conn, String indexTableNa
...
-            conn.createStatement().execute("CREATE INDEX " + indexTableName + " on " +
-                    dataTableName + " (val1) include (val2, val3)");
-            // Configure IndexRegionObserver to fail the last write phase (i.e., the post index update phase)
-            IndexRegionObserver.setFailPostIndexUpdatesForTesting(true);
-            conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
-            conn.commit();
-            IndexRegionObserver.setFailPostIndexUpdatesForTesting(false);
-            // check that in the first phase we don't send the full row.
-            // We count the num of cells for this
-            checkUnverifiedCellCount(conn, indexTableName);
```
[the remark this answers is not in the document]
Comment (kadirozde) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java]: Ok. I will do that.

File: phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesOptions.java (hunk 1 above)
Comment (kadirozde) [reply phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesOptions.java]: The team including Geoffrey and Vincent decided that 7 day delete time is long enough in practice to make sure a row will be replicated eventually.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kadirozde) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java]: Done

**Pull request closed without merging**